infra/main.bicep

```diff
@@ -146,47 +179,46 @@ module storage './core/storage/storage-account.bicep' = {
     networkAcls: {
       defaultAction: 'Deny'
     }
-    containers: [{name: uploadDataDeploymentStorageContainerName}, {name: orchestrateIngestionDeploymentStorageContainerName}]
+    containers: [{name: uploadDataDeploymentStorageContainerName}, {name: orchestrateIngestionDeploymentStorageContainerName}, {name: dataUploadContanerName}]
   }
 }
 
+var principalIds = [uploadData.outputs.SERVICE_API_IDENTITY_PRINCIPAL_ID, orchestrateIngestion.outputs.SERVICE_API_IDENTITY_PRINCIPAL_ID, principalId]
+
 //Storage Blob Data Owner role, Storage Blob Data Contributor role, Storage Table Data Contributor role
 // Allow access from apps to storage account using managed identity
 var storageRoleIds = ['b7e6dc6d-f1e8-4753-8033-0f276bb0955b', 'ba92f5b4-2d11-453d-a403-e96b0029c9fe', '0a9a7e1f-b9d0-4cc4-a60d-0319b160aaa3']
-var storagePrincipalIds = [uploadData.outputs.SERVICE_API_IDENTITY_PRINCIPAL_ID, orchestrateIngestion.outputs.SERVICE_API_IDENTITY_PRINCIPAL_ID, principalId]
 module storageRoleAssignments 'app/storage-Access.bicep' = [for roleId in storageRoleIds: {
   name: 'blobRole${roleId}'
   scope: rg
   params: {
     storageAccountName: storage.outputs.name
     roleId: roleId
-    principalIds: storagePrincipalIds
+    principalIds: principalIds
```

Comment:
Are you changing this because we're consolidating principals (as discussed, this morning)?

Reply:
This was more because both storage and service bus needed to have the roles assigned to all three principals (the two azure functions and the optional account of the user if AZURE_PRINCIPAL_ID variable is set on azd). You'll notice I reuse the principalIds for the service bus section too.